### DISCUSSION TITLE
Access to OpenAI o1-preview

### DISCUSSION THREAD

**Discussion by Saku-T** [community/community#140697] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I joined the waitlist for early access to OpenAI o1 on GitHub Models and received an email confirming that I gained access to the limited public beta for GitHub Models. However, I am unable to try the OpenAI o1-preview.
How can I access it?

**Comment by tdh1985** [comment]
Same, I dont get any additional chat mode to change to in either VSCode or insiders. Latest edition and extensions

**Comment by matthewisabel** [comment]
Hey @Saku-T and @tdh1985, do you mind letting me know what you see here? Will debug a bit on our end.

https://github.com/marketplace/models/azure-openai/o1-preview/playground

**Reply by matthewisabel** [reply]
We're still rolling out o1 access and you can sign up here: https://github.com/o1-waitlist-signup

**Reply by Saku-T** [reply]
Thanks for the reply.
I've already signed up for the OpenAI o1 waitlist and received an email confirming that I have access to the limited beta of GitHub Models.
However, I can't access the o1 preview playground.
![スクリーンショット 2024-10-08 234458](https://github.com/user-attachments/assets/8d753cff-9263-4ef9-889d-44349dd44f60)
![スクリーンショット 2024-10-08 234609](https://github.com/user-attachments/assets/eae5c683-d325-4172-9045-d4d9150eff86)
![スクリーンショット 2024-10-08 234743](https://github.com/user-attachments/assets/614c1e1c-a51c-4916-99fc-e916ea36c7b5)

**Reply by tdh1985** [reply]
> Hey @Saku-T and @tdh1985, do you mind letting me know what you see here? Will debug a bit on our end.
> 
> https://github.com/marketplace/models/azure-openai/o1-preview/playground

I get a 404 error, 
![image](https://github.com/user-attachments/assets/b12c0f76-85cc-48f8-9f31-2a976d392da8)

**Reply by Saku-T** [reply]
I get a 404 error too.

**Reply by niknshinde** [reply]
hey can you try now cause i also facing same issue but now i get access to most of the model

**Comment by ganxinyu666** [comment]
me too

**Comment by LeeXDA21** [comment]
me too but on the page its asking me to join the waitlist.

**Reply by LeeXDA21** [reply]
![Screenshot_20241008_184822_Gmail](https://github.com/user-attachments/assets/b50cfa2c-5214-4256-b41f-066bee7274dd)
It says on the waitlist page we will receive an email with setup instructions but I don't see any setup instructions provided?

**Comment by Neoony** [comment]
I guess in copilot chat extension, go into chat
![image](https://github.com/user-attachments/assets/7bbe3765-c00a-41b5-9bd1-6c670098c98f)

it was default 4o for me when I opened it

You have to be on "VS Code Insiders v1.94 and the pre-release Copilot extension"

not sure whether this means that the code completions are not 4o, only the chat?

well this above is only for VScode

**Reply by Neoony** [reply]
If I try https://github.com/marketplace/models/azure-openai/o1-preview/playground
It goes to "apply for waitlist"

so yeah this is something different (never saw that waitlist)

But I got 4o in vscode

**Reply by Saku-T** [reply]
I also installed VS Code Insiders v1.95 and the pre-release Copilot extension, but the model selection doesn't appear.
Do I need to do any additional settings?
![スクリーンショット 2024-10-09 091255](https://github.com/user-attachments/assets/e5f64177-1c1b-497f-b4ba-e72c98d25d1b)

**Reply by SantiagoStrauss** [reply]
@Saku-T do you still have that problem?

**Reply by Saku-T** [reply]
Yes, I do.

**Reply by sbhat-ppinc** [reply]
Same problem and I see a 404

**Comment by tdh1985** [comment]
Same

**Comment by Akash1134** [comment]
Hi All,

Thank you to everyone who participated in this discussion about accessing OpenAI o1-preview models. We appreciate your patience as we've worked to address the issues reported in this thread.

## Current Status Update

Since this discussion began in October 2024, we've made significant progress with our OpenAI model rollouts:

1. **Full Availability**: OpenAI o1 models are now generally available to all GitHub Copilot users with eligible subscriptions, resolving the access issues reported in this thread.

2. **Model Selection Interface**: The model selection dropdown in VS Code and other supported IDEs has been updated to properly display all available models, including o1-mini, o1, and other variants.

3. **GitHub Marketplace Experience**: The 404 errors previously encountered when trying to access o1-preview through the GitHub Marketplace have been resolved.

## How to Access o1 Models Today

To ensure you have access to OpenAI o1 models:

1. **Update Your Extensions**: Make sure you're using the latest version of GitHub Copilot and GitHub Copilot Chat extensions in your IDE.

2. **Check Your Settings**: Visit [https://github.com/settings/copilot](https://github.com/settings/copilot) to verify that the models you want to use are enabled.

3. **Model Selection**: In VS Code, click the GitHub Copilot icon in the status bar or use the model selector dropdown in the Copilot Chat panel to choose your preferred model.

4. **Restart Your IDE**: Sometimes a complete restart is needed after updating extensions or changing settings.

## Known Limitations

- Some JetBrains IDE users may still experience limited model availability compared to VS Code.
- Enterprise users may need administrator approval for o1 model access.

If you're still experiencing issues accessing o1 models after following these steps, please start a new discussion with current details, including your Copilot subscription type and IDE versions.

We're marking this thread as resolved given the time that has passed and the general availability of these models. Thank you for your feedback and patience throughout this process.
